### PR TITLE
feat: [P4PU-628] added tf scripts for allarm

### PR DESCRIPTION
**Pull request by oleksiybozhykntt**
### List of changes
added tf scripts for allarm

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [ ] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/06_domains/cittadini-app/00_keyvault.tf
[the remark this answers is not in the document]
Comment (oleksiybozhykntt) [reply src/06_domains/cittadini-app/00_keyvault.tf:1]: removed

File: src/06_domains/cittadini-app/00_data.tf [hunk 1]
```diff
@@ -43,3 +43,8 @@ data "azurerm_key_vault_secret" "appinsights_connection_string" {
   name         = "appinsights-connection-string"
   key_vault_id = local.kv_domain_id
 }
+
+data "azurerm_log_analytics_workspace" "log_analytics" {
+  name                = var.log_analytics_workspace_name
+  resource_group_name = var.log_analytics_workspace_resource_group_name
```
[the remark this answers is not in the document]
Comment (oleksiybozhykntt) [reply src/06_domains/cittadini-app/00_data.tf]: updated with locals and removed varibales

File: src/06_domains/cittadini-app/99_locals.tf [hunk 2]
```diff
@@ -40,4 +40,8 @@ locals {
   monitor_resource_group_name     = "${local.project_core}-monitor-rg"
   monitor_action_group_slack_name = "SlackPagoPA"
   monitor_action_group_email_name = "PagoPA"
+
+  azurerm_log_analytics_workspace_log_analytics_name = "${local.project_core}-law"
+  azurerm_log_analytics_workspace_log_analytics_resource_group_name = "${local.project_core}-monitor-rg"
```
Comment (antonioT90) [inline src/06_domains/cittadini-app/99_locals.tf:45]: let's keep simpler these names
```suggestion
  log_analytics_name = "${local.project_core}-law"
  log_analytics_resource_group_name = "${local.project_core}-monitor-rg"
```

Comment (antonioT90) [reply src/06_domains/cittadini-app/99_locals.tf:45]: moreover, the monitor resource group is already referenced:
see data.azurerm_resource_group.monitor_rg"

File: src/06_domains/cittadini-app/99_variables.tf [hunk 3]
```diff
@@ -93,3 +93,7 @@ variable "argocd_server_addr" {
   description = "ArgoCD hostname"
 }
 
+variable "arc_alert_enabled" {
```
Comment (antonioT90) [inline src/06_domains/cittadini-app/99_variables.tf:96]: in UAT  environment configuration this should be true (src/06_domains/cittadini-app/env/itn-uat/terraform.tfvars)

**Review by antonioT90 — CHANGES_REQUESTED** [review]

**Review by antonioT90 — APPROVED** [review]

File: src/06_domains/cittadini-app/12_alert.tf [hunk 4]
```diff
@@ -0,0 +1,68 @@
+resource "azurerm_monitor_action_group" "slackArc" {
```
Comment (umbcoppolabottazzi) [inline src/06_domains/cittadini-app/12_alert.tf:1]: Please change the resource name to kebab case

Comment (umbcoppolabottazzi) [reply src/06_domains/cittadini-app/12_alert.tf:1]:
```suggestion
resource "azurerm_monitor_action_group" "slack_cittadini" {
```

File: src/06_domains/cittadini-app/12_alert.tf [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+  count = var.arc_alert_enabled ? 1 : 0
+
+  name                = "SlackArc"
```
Comment (umbcoppolabottazzi) [inline src/06_domains/cittadini-app/12_alert.tf:5]:
```suggestion
  name                = "slack-${var.domain}"
```

**Comment by umbcoppolabottazzi** [comment]
Great Job!

**Review by umbcoppolabottazzi — APPROVED** [review]

**Pull request merged**